backend/capellacollab/users/tokens/crud.py

```diff
@@ -0,0 +1,66 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import datetime
+from collections import abc
+
+import argon2
+import sqlalchemy as sa
+from sqlalchemy import orm
+
+from capellacollab.core import credentials
+
+from . import models
+
+
+def create_token(
+    db: orm.Session, user_id: int, description: str
+) -> tuple[models.DatabaseUserTokenModel, str]:
+    password = credentials.generate_password(32)
+    ph = argon2.PasswordHasher()
+    token_data = models.DatabaseUserTokenModel(
+        user_id=user_id,
+        hash=ph.hash(password),
+        expiration_date=datetime.datetime.now() + datetime.timedelta(days=30),
+        description=description,
+    )
+    db.add(token_data)
+    db.commit()
+    return token_data, password
+
+
+def get_token(
+    db: orm.Session, password: str, user_id: int
+) -> models.DatabaseUserTokenModel | None:
+    ph = argon2.PasswordHasher()
+    token_list = get_token_by_user(db, user_id)
+    if token_list:
+        for token in token_list:
+            try:
+                ph.verify(token.hash, password)
+                return token
+            except argon2.exceptions.VerifyMismatchError:
+                pass
+    return None
+
+
+def get_token_by_user(
+    db: orm.Session, user_id: int
+) -> abc.Sequence[models.DatabaseUserTokenModel] | None:
+    return (
+        db.execute(
+            sa.select(models.DatabaseUserTokenModel).where(
+                models.DatabaseUserTokenModel.user_id == user_id
+            )
+        )
+        .scalars()
+        .all()
+    )
+
+
+def delete_token(
+    db: orm.Session, existing_token: models.DatabaseUserTokenModel
+) -> models.DatabaseUserTokenModel:
+    db.delete(existing_token)
+    db.commit()
+    return existing_token
```

Comment:
```suggestion
```
We just deleted the token, so I would not return it here.

Reply:
I was thinking about it whether a programming paradigm exists but I don't know. I have often times seen functions that delete objects return the deleted object. That meaning that deletion was successful. Do you have a strong opinion on that?

Reply:
For me we have two cases here. Either the deletion fails and then the `db.delete(...)` should raise an exception (i.e., we would never come to the return anyways) or everything works. So I don't really see a use case for returning the deleted object. How do the places where you've seen this handle this?

Reply:
I deleted it